x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/expression/FieldAttribute.java

```diff
@@ -132,7 +132,7 @@ protected Attribute clone(
         boolean synthetic
     ) {
         FieldAttribute qualifiedParent = parent != null ? (FieldAttribute) parent.withQualifier(qualifier) : null;
-        return new FieldAttribute(source, qualifiedParent, name, field, qualifier, nullability, id, synthetic);
+        return new FieldAttribute(source, qualifiedParent, name, type, field, qualifier, nullability, id, synthetic);
```

Comment:
QL change: Previously, `FieldAttribute`'s data type was always the same as the contained `EsField`'s.

Reply:
Actually, I realized that's not true - one can already set a different data type in the constructor of `FieldAttribute`; but it will be lost the next time the `FieldAttribute` is `clone`d, which may be regarded as wrong behavior of `clone` and would be fixed here.